spec/operators/takeLast-spec.ts

```diff
@@ -14,6 +14,46 @@ describe('Observable.prototype.takeLast()', () => {
     expectSubscriptions(e1.subscriptions).toBe(e1subs);
   });
 
+  it('should take last three values', (done: DoneSignature) => {
+    const expected = [2, 3, 4];
+
+    Observable.of(0, 1, 2, 3, 4).takeLast(3).subscribe((x) => {
+      expect(x).toBe(expected.shift());
+    }, done.fail, () => {
+      expect(expected.length).toBe(0);
+      done();
+    });
+  });
+
+  it('should take all element when try to take larger then source', (done: DoneSignature) => {
+    const expected = [0, 1, 2];
+
+    Observable.of(0, 1, 2).takeLast(5).subscribe((x) => {
+      expect(x).toBe(expected.shift());
+    }, done.fail, () => {
+      expect(expected.length).toBe(0);
+      done();
+    });
+  });
+
+  it('should take all element when try to take exact', (done: DoneSignature) => {
+    const expected = [0, 1, 2];
+
+    Observable.of(0, 1, 2).takeLast(3).subscribe((x) => {
+      expect(x).toBe(expected.shift());
+    }, done.fail, () => {
+      expect(expected.length).toBe(0);
+      done();
+    });
+  });
```

Comment:
I don't mind keeping one synchronous test, of course, but the marble tests are more analogous to the issue that was reported (which was using an interval).